### ISSUE TITLE
bug - cli shouldn't use internal domains as default domains

### ISSUE THREAD

**Issue by ameowlia** [cloudfoundry/cli#1420]
## What occurred

When the CLI picks a domain to use for the default domain during `cf push` it picks the first domain that the cc returns. Even if it is an internal domain. `cf push` then makes internal routes instead of external routes. 

## Steps to Reproduce

```
# delete all your domains 
# create an internal domain 
cf curl /v2/shared_domains -X POST -d  '{"name":"apps.internal", "internal":true}'

# create an external domain
cf create-shared-domain bosh-lite.com

# push an app
cf push MYAPP

# try to curl the app 
curl myapp.apps.internal
# it fails because it used an internal domain :( 

```

## What you expected to occur

The CLI should pick the first NON internal domain. 

CAPI 1.61.0+ returns an `internal` boolean property.

```
{
  "total_results": 2,
  "total_pages": 1,
  "prev_url": null,
  "next_url": null,
  "resources": [
    {
      "metadata": {
        "guid": "5895c770-14d5-4b1e-9e36-8810f4211724",
        "url": "/v2/shared_domains/5895c770-14d5-4b1e-9e36-8810f4211724",
        "created_at": "2018-07-18T01:30:40Z",
        "updated_at": "2018-07-18T01:30:40Z"
      },
      "entity": {
        "name": "apps.internal",
        "internal": true,
        "router_group_guid": null,
        "router_group_type": null
      }
    },
    {
      "metadata": {
        "guid": "ee0acb82-ac2f-451a-ad1a-51aaca65e6ce",
        "url": "/v2/shared_domains/ee0acb82-ac2f-451a-ad1a-51aaca65e6ce",
        "created_at": "2018-07-18T01:33:57Z",
        "updated_at": "2018-07-18T01:33:57Z"
      },
      "entity": {
        "name": "bosh-lite.com",
        "internal": false,
        "router_group_guid": null,
        "router_group_type": null
      }
    }
  ]
}
``` 

## CLI Version

cf version 6.37.0+a40009753.2018-05-25

## CC API Endpoint Version

This affects CAPI Release version: 1.61.0+
CF Networking Version: 2.5.0+

## Any other relevant information

We want to seed internal domains in CAPI, so that fresh deployments are able to run CATs tests without a shim or extra script. However, seeding an internal domain has the side affect of making them the top domain entry. 

This issue is blocking us from providing a seamless user experience for service discovery users. One customer has already run into this issue on their own.

**Comment by ameowlia** [comment]
## CF Networking Prioritization

We've been creating a lot of issues with you all to cover our current and upcoming work. I just wanted to share how we have prioritized these stories within our team. 

1. BUG: CLI shouldn't use internal domains as the default external domain: https://github.com/cloudfoundry/cli/issues/1420
1. FEATURE: app to app policy within a space: https://github.com/cloudfoundry/cli/issues/1419
1. FEATURE: cli support for internal domains https://github.com/cloudfoundry/cli/issues/1423 (this issue)
1. FEATURE: support for egress policies (no issue created yet, but we will be working on this soon)

**Comment by ameowlia** [comment]
This is still important for us, and more customers have reported running into it. Yes there is a work around, but it is causing a bad `cf push` experience.

**Comment by pvaramballypivot** [comment]
@abbyachau 
Here's one that was reported this month
https://pivotal.slack.com/archives/C064Q36JG/p1533672727000292
There is one more from way back that I am unable to find. That is the one that triggered this github issue. I will add it here when I find it.

**Comment by abbyachau** [comment]
This bug was fixed in [CLI v6.39.0](https://github.com/cloudfoundry/cli/releases/tag/v6.39.0). Thanks for creating this issue @ameowlia!
